torchvision/models/vision_transformer.py

```diff
@@ -288,18 +288,11 @@ def _vision_transformer(
 ) -> VisionTransformer:
     if weights is not None:
         _ovewrite_named_param(kwargs, "num_classes", len(weights.meta["categories"]))
-        if isinstance(weights.meta["size"], int):
-            _ovewrite_named_param(kwargs, "image_size", weights.meta["size"])
-        elif isinstance(weights.meta["size"], Sequence):
-            if len(weights.meta["size"]) != 2 or weights.meta["size"][0] != weights.meta["size"][1]:
-                raise ValueError(
-                    f'size: {weights.meta["size"]} is not valid! Currently we only support a 2-dimensional square and width = height'
-                )
-            _ovewrite_named_param(kwargs, "image_size", weights.meta["size"][0])
-        else:
+        if weights.meta["min_size"][0] != weights.meta["min_size"][1]:
             raise ValueError(
-                f'weights.meta["size"]: {weights.meta["size"]} is not valid, the type should be either an int or a Sequence[int]'
```

Comment:
Replace `size` with `min_size` which stores the same info in ViT. I also simplify the logic here because all the extra checks are not needed. We fully control which weights get deployed and putting the wrong info in will cause issues anyway on our existing tests.